src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java

```diff
@@ -25,20 +31,42 @@
     private static final String OTHER_LIABILITIES_OR_ASSETS_TOTAL_ASSETS_LESS_CURRENT_LIABILITIES_PATH = OTHER_LIABILITIES_OR_ASSETS_PATH + ".total_assets_less_current_liabilities";
     private static final String OTHER_LIABILITIES_OR_ASSETS_TOTAL_NET_ASSETS_PATH = OTHER_LIABILITIES_OR_ASSETS_PATH + ".total_net_assets";
     private static final String CURRENT_ASSETS_TOTAL_PATH = BALANCE_SHEET_PATH + ".current_assets.total";
+    private static final String TOTAL_SHAREHOLDER_FUNDS_PATH = BALANCE_SHEET_PATH + ".capital_and_reserves.total_shareholder_funds";
 
     public Errors validateCurrentPeriod(@Valid CurrentPeriod currentPeriod) {
 
         Errors errors = new Errors();
 
+        validateTotalFixedAssets(currentPeriod, errors);
+        validateTotalOtherLiabilitiesOrAssets(currentPeriod, errors);
+        validateTotalCurrentAssets(currentPeriod, errors);
+        validateTotalShareholderFunds(currentPeriod, errors);
+
+        return errors;
+    }
+
+    public void validateTotalShareholderFunds(CurrentPeriod currentPeriod, Errors errors) {
 
-            validateTotalFixedAssets(currentPeriod, errors);
+        CapitalAndReserves capitalAndReserves = currentPeriod.getBalanceSheet().getCapitalAndReserves();
 
-            validateTotalOtherLiabilitiesOrAssets(currentPeriod, errors);
+        if (capitalAndReserves != null) {
 
-            validateTotalCurrentAssets(currentPeriod, errors);
+            Long calledUpShareCapital = Optional.ofNullable(capitalAndReserves.getCalledUpShareCapital()).orElse(0L);
+            Long sharePremiumAccount = Optional.ofNullable(capitalAndReserves.getSharePremiumAccount()).orElse(0L);
+            Long otherReserves = Optional.ofNullable(capitalAndReserves.getOtherReserves()).orElse(0L);
+            Long profitAndLoss = Optional.ofNullable(capitalAndReserves.getProfitAndLoss()).orElse(0L);
 
+            Long totalShareholderFunds = Optional.ofNullable(capitalAndReserves.getTotalShareholdersFund()).orElse(0L);
+            Long calculatedTotal = calledUpShareCapital + otherReserves + sharePremiumAccount + profitAndLoss;
+            validateAggregateTotal(totalShareholderFunds, calculatedTotal, TOTAL_SHAREHOLDER_FUNDS_PATH, errors);
 
-        return errors;
+            // Total shareholder funds must equal total net assets
+            Long totalNetAssets = currentPeriod.getBalanceSheet().getOtherLiabilitiesOrAssets().getTotalNetAssets();
+            if (totalNetAssets != totalShareholderFunds) {
```

Comment:
You are comparing objects and need to use `!totalNetAssets.equals(totalShareholderFunds)`